python/cudf/cudf/core/dataframe.py

```diff
@@ -2744,11 +2744,14 @@ def reindex(
         else:
             if columns is None:
                 columns = labels
-        df = (
-            self
-            if columns is None
-            else self[list(set(self._column_names) & set(columns))]
-        )
+        if columns is None:
+            df = self
+        else:
+            columns = as_index(columns)
+            intersection = self._data.to_pandas_index().intersection(
```

Comment:
Happy to have suggestion if there's a way to do this without dispatching to pandas. This was to ensure the MultiIndex intersection case worked correctly